### PR TITLE
Wadeea react homework week2

### PR DESCRIPTION
**Pull request by WadeeaKiwan**

### CODE REVIEW

File: homework-submission/W2-homework/todo-app/src/Components/AddItem/AddItem.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    this.setState({
+      [e.target.id]: e.target.value,
+    });
+  };
+
+  handleSubmit = e => {
+    e.preventDefault();
+    if (e.target.description.value === '') {
+      return false;
+    } else {
```
Comment (Meno-101) [inline homework-submission/W2-homework/todo-app/src/Components/AddItem/AddItem.js]: If you are returning no need for the `else` close!

the return keyword stops the function in it's exact place and never execute any line after.

File: homework-submission/W2-homework/todo-app/src/App.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    let todoItems = this.state.todoItems.filter(todoItem => {
+      return todoItem.id !== id;
+    });
+    this.setState({ todoItems });
+  };
+
+  addTodoItem = todoItem => {
+    let todoItems = this.state.todoItems;
+    todoItem.id = Math.random() * 10 + 1;
+    todoItems.push(todoItem);
```
Comment (Meno-101) [inline homework-submission/W2-homework/todo-app/src/App.js]: You are manipulating the state directly!

If you tried the following:
```javascript
    console.log(this.state.todoItems.length) // if this 3
    todoItems.push(todoItem)
    console.log(this.state.todoItems.length) // this is changed to be 4, which it should not do that
```

the `.setState` method is on top of a complete lifecycle on behalf of eyes which made react makers make it asynchronous, and that is why the strange behavior you have saw here.

This is an important thing to keep in mind avoiding Bugs in the future, and understanding how react dealing with it's own state.

you could solve this as the following allowing your self to add an item without touching the state directly:
```javascript
this.setState({
  todoItems: [
    ...this.state.todoItems,
    {
      ...todoItem,
      id: /* Some unique id... */
    }
  ]
})
```

File: homework-submission/W2-homework/todo-app/src/Components/AddItem/AddItem.js (hunk 1 above)
Comment (WadeeaKiwan) [reply homework-submission/W2-homework/todo-app/src/Components/AddItem/AddItem.js]: Thanks... done!

File: homework-submission/W2-homework/todo-app/src/App.js (hunk 2 above)
Comment (WadeeaKiwan) [reply homework-submission/W2-homework/todo-app/src/App.js]: Many thanks for this advice.

**Comment by WadeeaKiwan** [comment]
> Excellent using of `create-react-app`!
> 
> I don't have much to say, I just noticed a one thing can be improved, no other words!
> 
> **Keep it up...**

Many thanks @Meno-101 for your feedback and advices.

**Pull request closed without merging**